src/modules/clustering/hierarchical_clustering_method.cpp

```diff
@@ -87,25 +91,54 @@ std::vector<std::vector<Junction>> split_partition_based_on_mate2(std::vector<Ju
     return splitted_partition;
 }
 
-int junction_distance(Junction const & lhs, Junction const & rhs)
+double junction_distance(Junction const & lhs, Junction const & rhs)
 {
+    // lhs and rhs connect the same chromosomes with the same orientations
     if ((lhs.get_mate1().seq_name == rhs.get_mate1().seq_name) &&
         (lhs.get_mate1().orientation == rhs.get_mate1().orientation) &&
         (lhs.get_mate2().seq_name == rhs.get_mate2().seq_name) &&
         (lhs.get_mate2().orientation == rhs.get_mate2().orientation))
     {
-        // Reference:                      ................
-        // Junction 1 with mates A and B:     A------->B    (2bp inserted)
-        // Junction 2 with mates C and D:    C------>D      (5bp inserted)
-        // Distance = 1 (distance A-C) + 2 (distance B-D) + 0 (no tandem duplication) + 3 (absolute insertion size difference)
-        return (std::abs(lhs.get_mate1().position - rhs.get_mate1().position) +
-                std::abs(lhs.get_mate2().position - rhs.get_mate2().position) +
-                std::abs((int)(lhs.get_tandem_dup_count() - rhs.get_tandem_dup_count())) +
-                std::abs((int)(lhs.get_inserted_sequence().size() - rhs.get_inserted_sequence().size())));
+        // lhs and rhs are intra-chromosomal adjacencies
+        if (lhs.get_mate1().seq_name == lhs.get_mate2().seq_name)
+        {
+            // the directed size is the directed distance between both mates
+            // the directed size is positive for insertions and negative for deletions
+            int32_t lhs_directed_size = lhs.get_inserted_sequence().size() +
+                                        lhs.get_mate1().position -
+                                        lhs.get_mate2().position;
+            int32_t rhs_directed_size = rhs.get_inserted_sequence().size() +
+                                        rhs.get_mate1().position -
+                                        rhs.get_mate2().position;
+            // lhs and rhs have the same type (either deletion/inversion or insertion)
+            if ((lhs_directed_size < 0 && rhs_directed_size < 0) ||
+                (lhs_directed_size > 0 && rhs_directed_size > 0))
+            {
+                double position_distance = std::abs(lhs.get_mate1().position - rhs.get_mate1().position) / 1000.0;
+                // TODO (irallia 01.09.2021): std::abs((int)(lhs.get_tandem_dup_count() - rhs.get_tandem_dup_count()))
```

Comment:
What about this todo?

Reply:
Thats a todo for my next tandem duplication PR :)